### PR TITLE
Respect 'soft_fail' argument when running BatchSensors

### PR DESCRIPTION
**Pull request by utkarsharma2**
This PR intends to solve the issue that the soft_fail argument in BatchJobQueueSensor, BatchComputeEnvironmentSensor, and BatchSensor is not respected.

### CODE REVIEW

File: tests/providers/amazon/aws/sensors/test_batch.py [hunk 1]
```diff
@@ -174,6 +201,42 @@ def test_poke_invalid(
...
+        batch_compute_environment_sensor: BatchComputeEnvironmentSensor,
+        soft_fail,
+        expected_exception,
+    ):
+        compute_env = [{"status": "unknown_status"}]
+        mock_batch_client.describe_compute_environments.return_value = {"computeEnvironments": compute_env}
+        message = "AWS Batch compute environment failed. AWS Batch compute environment status:"
+        batch_compute_environment_sensor.soft_fail = soft_fail
+        with pytest.raises(expected_exception, match=message):
+            batch_compute_environment_sensor.poke({})
```
Comment (Lee-W) [inline tests/providers/amazon/aws/sensors/test_batch.py]:
```suggestion
    @pytest.mark.parametrize(
        "soft_fail, expected_exception", ((False, AirflowException), (True, AirflowSkipException))
    )
    @pytest.mark.parametrize(
        "compute_env, error_message",
        (
            ([], "AWS Batch compute environment"),
            ([{"status": "unknown_status"}], "AWS Batch compute environment failed. AWS Batch compute environment status:")
        )
    )
    @mock.patch.object(BatchClientHook, "client")
    def test_fail_poke(
        self,
        mock_batch_client,
        batch_compute_environment_sensor: BatchComputeEnvironmentSensor,
        compute_env,
        error_message,
        soft_fail,
        expected_exception,
    ):
        compute_env = []
        mock_batch_client.describe_compute_environments.return_value = {"computeEnvironments": compute_env}
        message = 
        batch_compute_environment_sensor.soft_fail = soft_fail
        with pytest.raises(expected_exception, match=error_message):
            batch_compute_environment_sensor.poke({})
```

Not sure whether it makes sense, but I might try something like this

File: tests/providers/amazon/aws/sensors/test_batch.py [hunk 2]
```diff
@@ -242,3 +305,33 @@ def test_poke_invalid(self, mock_batch_client, batch_job_queue_sensor: BatchJobQ
...
+    def test_fail_poke_invalid_jobqueues_status(
+        self, mock_batch_client, batch_job_queue_sensor: BatchJobQueueSensor, soft_fail, expected_exception
+    ):
+        job_queue = [{"status": "UNKNOWN_STATUS"}]
+        mock_batch_client.describe_job_queues.return_value = {"jobQueues": job_queue}
+        batch_job_queue_sensor.treat_non_existing_as_deleted = False
+        batch_job_queue_sensor.soft_fail = soft_fail
+        message = "AWS Batch job queue"
+        with pytest.raises(expected_exception, match=message):
+            batch_job_queue_sensor.poke({})
```
Comment (Lee-W) [inline tests/providers/amazon/aws/sensors/test_batch.py]:
```suggestion
    @pytest.mark.parametrize(
        "soft_fail, expected_exception", ((False, AirflowException), (True, AirflowSkipException))
    )
    @pytest.mark.parametrize("job_queue", ([], [{"status": "UNKNOWN_STATUS"}]))
    @mock.patch.object(BatchClientHook, "client")
    def test_fail_poke(
        self, mock_batch_client, batch_job_queue_sensor: BatchJobQueueSensor, job_queue, soft_fail, expected_exception
    ):
        mock_batch_client.describe_job_queues.return_value = {"jobQueues": job_queue}
        batch_job_queue_sensor.treat_non_existing_as_deleted = False
        batch_job_queue_sensor.soft_fail = soft_fail
        message = "AWS Batch job queue"
        with pytest.raises(expected_exception, match=message):
            batch_job_queue_sensor.poke({})
```

same above

File: tests/providers/amazon/aws/sensors/test_batch.py (hunk 1 above)
Comment (utkarsharma2) [reply tests/providers/amazon/aws/sensors/test_batch.py]: It does, we can do this

File: tests/providers/amazon/aws/sensors/test_batch.py [hunk 3]
```diff
@@ -106,6 +106,33 @@ def test_execute_failure_in_deferrable_mode_with_soft_fail(self, deferrable_batc
...
+    @mock.patch.object(BatchClientHook, "get_job_description")
+    def test_fail_poke_unknown_state(
+        self, mock_get_job_description, batch_sensor: BatchSensor, soft_fail, expected_exception
+    ):
+        state = "unknown_state"
+        mock_get_job_description.return_value = {"status": state}
+        message = f"Batch sensor failed. Unknown AWS Batch job status: {state}"
+        batch_sensor.soft_fail = soft_fail
+        with pytest.raises(expected_exception, match=message):
+            batch_sensor.poke({})
```
Comment (Lee-W) [inline tests/providers/amazon/aws/sensors/test_batch.py]:
```suggestion
    @pytest.mark.parametrize(
        "soft_fail, expected_exception", ((False, AirflowException), (True, AirflowSkipException))
    )
    @pytest.mark.parametrize(
        "state, error_message",
        (
            (BatchClientHook.FAILURE_STATE, f"Batch sensor failed. AWS Batch job status: {BatchClientHook.FAILURE_STATE}"),
            ("unknown_state", "Batch sensor failed. Unknown AWS Batch job status: unknown_state")
        )
    )
    @mock.patch.object(BatchClientHook, "get_job_description")
    def test_fail_poke(
        self, mock_get_job_description, batch_sensor: BatchSensor, status, error_message, soft_fail, expected_exception
    ):
        mock_get_job_description.return_value = {"status": state}
        batch_sensor.soft_fail = soft_fail
        with pytest.raises(expected_exception, match=error_message):
            batch_sensor.poke({})
```

Forgot this one

**Review by uranusjr — APPROVED** [review]

**Review by vincbeck — APPROVED** [review]

**Pull request merged**